### ISSUE TITLE
Revise document with inconsistency

### ISSUE THREAD

**Issue by npuichigo** [PyO3/pyo3#3639]
Currently there're many inconsistency code snippets in document, for example the example output is not consistent with example code https://pyo3.rs/v0.20.0/function/signature

**Comment by adamreichold** [comment]
Since you already checked that, maybe you propose a pull request to resolve the issue? Thanks!

**Comment by npuichigo** [comment]
I'll collect the issues and fix in one PR

**Comment by luxedo** [comment]
For this specific case I think it was solved in https://github.com/PyO3/pyo3/commit/a200a5703edcfd43b8dad887c573449cc623f487

0.20
<img width="904" height="393" alt="Image" src="https://github.com/user-attachments/assets/85989c78-abc0-4b09-ad7a-3b1ca32d89af" />

0.28
<img width="968" height="366" alt="Image" src="https://github.com/user-attachments/assets/95fdd828-7df0-48d7-acd7-8b3433a724f8" />

**Comment by davidhewitt** [comment]
Thanks @luxedo, let's call this resolved in that case!

**Comment by luxedo** [comment]
I'm not yet familiar with the whole development pipeline, but so far I have found the doctests for Rust only. Is there any doctests for python code? That would easily detect those mistakes. We could even have a crate/package for defining custom behavior like this https://github.com/scipy/scipy_doctest

**Comment by davidhewitt** [comment]
I don't think we have any Python doctests. I am open to ideas, as long as the CI cost is not too high!
